internal/common/db.go

```diff
@@ -35,6 +35,9 @@ type Cluster struct {
 
 	// The ID of the subscription created in AMS
 	AmsSubscriptionID strfmt.UUID `json:"ams_subscription_id"`
+
+	// ImageGenerated indicates if the discovery image was generated successfully.
+	ImageGenerated bool `json:"image_generated"`
```

Comment:
@yevgeny-shnaidman @avishayt  How should this attribute be populated when assisted-service is upgraded ?
If it will be initialized by its default value (false), it will result in regenerating the discovery ISO once per cluster (even when not needed)
Setting it to 'true' cannot reflect the actual status of the image (don't think it will be possible to check the image availability from the migration, nor if its generation resulted in the desired result)

If the cost of having it set to 'false' isn't too high, I'd prefer to leave it that way.

Reply:
False is fine IMO
But where is the API definition?  I think it would be better to have a "state" property for the image.

Reply:
Why should we expose it in the API ? Its purpose is to indicate internally if the image was generated correctly or if it should be re-generated.
From user perspective, the generate-iso  / d/w iso API should remain the same (but produce the image in case it wasn't successfully generated)

Reply:
I see.  Its usage wasn't clear to me.

Reply:
I'll clarify it on the attribute documentation